### PR TITLE
docs: translation for `Component(class)`

### LINKED ISSUE #548: `Component`
Page: https://id.react.dev/reference/react/Component

### PR DESCRIPTION
**Pull request by aroyan**
Closes #548 

## Description

Translate the Component page.
Page URL: https://id.react.dev/reference/react/Component

### CODE REVIEW

**Comment by aroyan** [comment]
Please review @mhaidarhanif @zainfathoni @mazipan @resir014

File: src/content/reference/react/Component.md [hunk 1]
```diff
@@ -4,18 +4,18 @@ title: Component
 
 <Pitfall>
 
-We recommend defining components as functions instead of classes. [See how to migrate.](#alternatives)
+Kami merekomendasikan mendefinisikan komponen sebagai fungsi dari pada kelas. [Lihat bagaimana cara migrasi.](#alternatives)
```
Comment (mhaidarhanif) [inline src/content/reference/react/Component.md:7]:
```suggestion
Kami merekomendasikan mendefinisikan komponen sebagai fungsi daripada kelas. [Lihat bagaimana cara migrasi.](#alternatives)
```

https://kbbi.web.id/daripada

File: src/content/reference/react/Component.md [hunk 2]
```diff
@@ -4,18 +4,18 @@ title: Component
...
 
 </Pitfall>
 
 <Intro>
 
-`Component` is the base class for the React components defined as [JavaScript classes.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes) Class components are still supported by React, but we don't recommend using them in new code.
+`Component` adalah kelas dasar untuk komponen React yang didefinisikan segabai [JavaScript classes.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes) Kelas komponen masih disuport oleh React, tetapi kami tidak merekomendasikan untuk menggunakannya di kode baru.
```
Comment (mhaidarhanif) [inline src/content/reference/react/Component.md:13]:
```suggestion
`Component` adalah kelas dasar untuk komponen React yang didefinisikan sebagai [JavaScript classes.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes) Kelas komponen masih didukung oleh React, tetapi kami tidak merekomendasikan untuk menggunakannya di kode baru.
```

**Review by mhaidarhanif — CHANGES_REQUESTED** [review]
Because it seems there are some typos, please check again.

Karena sepertinya masih ada beberapa _typo_, silakan dicek lagi dahulu.

File: src/content/reference/react/Component.md [hunk 3]
```diff
@@ -109,17 +109,17 @@ Reading `this.props` in class components is equivalent to [declaring props](/lea
 
 <Deprecated>
 
-This API will be removed in a future major version of React. [Use `createRef` instead.](/reference/react/createRef)
+API ini akan dihapus pada versi utama React di masa depan. [Gunakan `createRef` sebagai gantinya.](/reference/react/createRef)
```
Comment (resir014) [inline src/content/reference/react/Component.md:112]:
```suggestion
API ini akan dihapus pada versi mayor React di masa depan. [Gunakan `createRef` sebagai gantinya.](/reference/react/createRef)
```

`versi utama` -> `versi mayor`

File: src/content/reference/react/Component.md [hunk 4]
```diff
@@ -1824,15 +1826,15 @@ button { margin-left: 10px; }
 
 <Note>
 
-If your component does not synchronize with any external systems, [you might not need an Effect.](/learn/you-might-not-need-an-effect)
+Jika komponent Anda tidak disinkronkan dengan sistem eksternal apapun, [Anda mungkin tidak membutuhkan sebuah *Effect*.](/learn/you-might-not-need-an-effect)
```
Comment (resir014) [inline src/content/reference/react/Component.md:1829]:
```suggestion
Jika komponen Anda tidak disinkronkan dengan sistem eksternal apapun, [Anda mungkin tidak membutuhkan sebuah *Effect*.](/learn/you-might-not-need-an-effect)
```

komponent -> komponen

File: src/content/reference/react/Component.md [hunk 5]
```diff
@@ -671,41 +673,41 @@ You can also pass a function to `setState`. It lets you update state based on th
   }
 ```
 
-You don't have to do this, but it's handy if you want to update state multiple times during the same event.
+Anda tidak harus melakukan ini, tetapi akan berguna jika Anda ingin memperbarui *state* beberapa kali selama *event* yang sama.
 
 #### Parameters {/*setstate-parameters*/}
```
Comment (resir014) [inline src/content/reference/react/Component.md:678]:
```suggestion
#### Parameter {/*setstate-parameters*/}
```

File: src/content/reference/react/Component.md [hunk 6]
```diff
@@ -732,106 +734,106 @@ class Rectangle extends Component {
 
 ```
 
-React calls `shouldComponentUpdate` before rendering when new props or state are being received. Defaults to `true`. This method is not called for the initial render or when [`forceUpdate`](#forceupdate) is used.
+React memanggil `shouldComponentUpdate` sebelum *render*-ing ketika *props* baru atau *state* sedang diterima. Default `true`. Metode ini tidak dipanggil untuk *render* awal atau saat [`forceUpdate`](#forceupdate) digunakan.
 
 #### Parameters {/*shouldcomponentupdate-parameters*/}
```
Comment (resir014) [inline src/content/reference/react/Component.md:739]:
```suggestion
#### Parameter {/*shouldcomponentupdate-parameters*/}
```

File: src/content/reference/react/Component.md [hunk 7]
```diff
@@ -184,73 +184,73 @@ class Counter extends Component {
   }
 ```
 
-A constructor should not contain any side effects or subscriptions.
+Sebuah *constructor* sebaiknya tidak mengandung *side effects* atau *subscriptions*.
 
 #### Parameters {/*constructor-parameters*/}
```
Comment (resir014) [inline src/content/reference/react/Component.md:189]:
```suggestion
#### Parameter {/*constructor-parameters*/}
```

File: src/content/reference/react/Component.md [hunk 8]
```diff
@@ -732,106 +734,106 @@ class Rectangle extends Component {
...
 
-- To initialize state, declare [`state`](#state) as a class field or set `this.state` inside the [`constructor`.](#constructor)
-- If you need to run a side effect or set up a subscription, move that logic to [`componentDidMount`](#componentdidmount) instead.
+- Untuk menginisialisasi *state*, deklarasikan [`state`](#state) sebagai bidang kelas atau setel `this.state` di dalam [`konstruktor`.](#konstruktor)
+- Jika Anda perlu menjalankan efek samping atau menyiapkan langganan, pindahkan logika tersebut ke [`componentDidMount`](#componentdidmount) sebagai gantinya.
 
-[See examples of migrating away from unsafe lifecycles.](https://legacy.reactjs.org/blog/2018/03/27/update-on-async-rendering.html#examples)
+[Lihat contoh migrasi dari *unsafe lifecycles*.](https://legacy.reactjs.org/blog/2018/03/27/update-on-async-rendering.html#examples)
 
 #### Parameters {/*unsafe_componentwillmount-parameters*/}
```
Comment (resir014) [inline src/content/reference/react/Component.md:780]:
```suggestion
#### Parameter {/*unsafe_componentwillmount-parameters*/}
```

File: src/content/reference/react/Component.md [hunk 9]
```diff
@@ -732,106 +734,106 @@ class Rectangle extends Component {
...
+- Jika Anda perlu **"mereset" beberapa state saat prop berubah,** pertimbangkan untuk membuat komponen [fully controlled](https://legacy.reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-controlled-component) atau [fully uncontrolled with a key](https://legacy.reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key) sebagai gantinya.
+- Jika Anda perlu **"menyesuaikan" beberapa status saat prop berubah,** periksa apakah Anda dapat menghitung semua informasi yang diperlukan hanya dari *prop* selama *render*-ing. Jika tidak bisa, gunakan [`static getDerivedStateFromProps`](/reference/react/Component#static-getderivedstatefromprops) sebagai gantinya.
 
-[See examples of migrating away from unsafe lifecycles.](https://legacy.reactjs.org/blog/2018/03/27/update-on-async-rendering.html#updating-state-based-on-props)
+[Lihat contoh migrasi dari *unsafe lifecycles*.](https://legacy.reactjs.org/blog/2018/03/27/update-on-async-rendering.html#updating-state-based-on-props)
 
 #### Parameters {/*unsafe_componentwillreceiveprops-parameters*/}
```
Comment (resir014) [inline src/content/reference/react/Component.md:815]:
```suggestion
#### Parameter {/*unsafe_componentwillreceiveprops-parameters*/}
```

File: src/content/reference/react/Component.md [hunk 10]
```diff
@@ -840,39 +842,39 @@ Calling [`setState`](#setstate) inside `UNSAFE_componentWillReceiveProps` in a c
...
-- If you need to read some information from the DOM (for example, to save the current scroll position) so that you can use it in [`componentDidUpdate`](#componentdidupdate) later, read it inside [`getSnapshotBeforeUpdate`](#getsnapshotbeforeupdate) instead.
+- Jika Anda perlu menjalankan efek samping (misalnya, mengambil data, menjalankan animasi, atau menginisialisasi ulang langganan) sebagai respons terhadap perubahan *prop* atau *state*, pindahkan logika tersebut ke [`componentDidUpdate`](#componentdidupdate) sebagai gantinya.
+- Jika Anda perlu membaca beberapa informasi dari DOM (misalnya, untuk menyimpan posisi gulir saat ini) sehingga Anda dapat menggunakannya di [`componentDidUpdate`](#componentdidupdate) nanti, baca di dalam [`getSnapshotBeforeUpdate`](#getsnapshotbeforeupdate ) sebagai gantinya.
 
-[See examples of migrating away from unsafe lifecycles.](https://legacy.reactjs.org/blog/2018/03/27/update-on-async-rendering.html#examples)
+[Lihat contoh migrasi dari *unsafe lifecycles*.](https://legacy.reactjs.org/blog/2018/03/27/update-on-async-rendering.html#examples)
 
 #### Parameters {/*unsafe_componentwillupdate-parameters*/}
```
Comment (resir014) [inline src/content/reference/react/Component.md:852]:
```suggestion
#### Parameter {/*unsafe_componentwillupdate-parameters*/}
```

File: src/content/reference/react/Component.md [hunk 11]
```diff
@@ -998,45 +1000,45 @@ class Greeting extends React.Component {
...
-If you define `static getDerivedStateFromError`, React will call it when a child component (including distant children) throws an error during rendering. This lets you display an error message instead of clearing the UI.
+Jika Anda mendefinisikan `static getDerivedStateFromError`, React akan memanggilnya ketika komponen anak (termasuk anak jauh) melemparkan kesalahan selama *render*-ing. Ini memungkinkan Anda menampilkan pesan kesalahan alih-alih menghapus UI.
 
-Typically, it is used together with [`componentDidCatch`](#componentdidcatch) which lets you send the error report to some analytics service. A component with these methods is called an *error boundary.*
+Biasanya, ini digunakan bersama dengan [`componentDidCatch`](#componentDidCatch) yang memungkinkan Anda mengirim laporan kesalahan ke beberapa layanan analitik. Komponen dengan metode ini disebut *error boundary.*
 
-[See an example.](#catching-rendering-errors-with-an-error-boundary)
+[Lihat contoh.](#catching-rendering-errors-with-an-error-boundary)
 
 #### Parameters {/*static-getderivedstatefromerror-parameters*/}
```
Comment (resir014) [inline src/content/reference/react/Component.md:1017]:
```suggestion
#### Parameter {/*static-getderivedstatefromerror-parameters*/}
```

File: src/content/reference/react/Component.md [hunk 12]
```diff
@@ -1062,46 +1064,46 @@ class Form extends Component {
...
+- Jika Anda perlu **melakukan efek samping** (misalnya, pengambilan data atau animasi) sebagai respons terhadap perubahan *props*, gunakan metode [`componentDidUpdate`](#componentdidupdate) sebagai gantinya.
+- Jika Anda ingin **menghitung ulang beberapa data hanya saat prop berubah,** [gunakan pembantu memoisasi sebagai gantinya.](https://legacy.reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization)
+- Jika Anda ingin **"mereset" beberapa *state* saat *prop* berubah,** pertimbangkan untuk membuat komponen [fully controlled](https://legacy.reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-controlled-component) atau [fully uncontrolled with a key](https://legacy.reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key).
 
 </Pitfall>
 
 #### Parameters {/*static-getderivedstatefromprops-parameters*/}
```
Comment (resir014) [inline src/content/reference/react/Component.md:1079]:
```suggestion
#### Parameter {/*static-getderivedstatefromprops-parameters*/}
```

**Review by resir014 — APPROVED** [review]

**Comment by aroyan** [comment]
Terima kasih bang sudah dibantu fix @resir014

**Pull request merged**